tests-functional/conftest.py

```diff
@@ -125,14 +126,228 @@ def pytest_report_header(config):
     ]
 
 
-@pytest.fixture(scope="function", autouse=True)
+@pytest.fixture(scope="function", autouse=False)
 def close_status_backend_containers(request):
+    """
+    Fixture to automatically cleanup Status backend containers after each test.
+    Should be used ONLY for tests that do not use backend_factory or class_backend fixtures.
+
+    This fixture ensures that all Status backend containers are properly stopped,
+    logs are saved, and containers are removed to prevent resource leaks and
+    conflicts between tests.
+
+    How it works:
+    1. Yields immediately (runs before test execution)
+    2. After test completes, checks if container reuse is enabled
+    3. If reuse is disabled, stops all containers in option.statusgo_containers
+    4. Saves logs from each container for debugging
+    5. Removes containers to free up system resources
+    6. Clears the containers list
+
+    Usage:
+    # Automatic cleanup for all tests in a class
+    @pytest.fixture(autouse=True)
+    def setup_cleanup(self, close_status_backend_containers):
+        yield
+
+    # Manual cleanup for specific test
+    def test_something(self, close_status_backend_containers):
+        # test code here
+        pass
+
+    # Skip cleanup for tests that reuse containers
+    class TestReuseContainers:
+        reuse_container = True  # This will skip cleanup
+
+    Parameters:
+        request: pytest request object containing test metadata
+
+    Dependencies:
+        - option.statusgo_containers: Global list of active containers
+        - Container objects with stop(), save_logs(), remove() methods
+
+    Scope: function (runs once per test function)
+    Autouse: False (must be explicitly requested)
+    """
     yield
     if hasattr(request.node.instance, "reuse_container"):
         return
     for container in option.statusgo_containers:
-        container.stop()  # pyright: ignore[reportAttributeAccessIssue]
-        container.save_logs()  # pyright: ignore[reportAttributeAccessIssue]
-        container.remove()  # pyright: ignore[reportAttributeAccessIssue]
+        try:
+            if hasattr(container, "container") and container.container:  # type: ignore
+                container.stop()  # pyright: ignore[reportAttributeAccessIssue]
+                try:
+                    container.save_logs()  # pyright: ignore[reportAttributeAccessIssue]
+                except RuntimeError as e:
+                    if "Container is not initialized" in str(e):
+                        logging.warning(f"Container already stopped, skipping log save: {e}")
+                    else:
+                        raise
+                container.remove()  # pyright: ignore[reportAttributeAccessIssue]
```

Comment:
Early return pattern makes things more readable 🙌 
```suggestion
            if not hasattr(container, "container") or not container.container:  # type: ignore
                continue
            container.stop()  # pyright: ignore[reportAttributeAccessIssue]
            try:
                container.save_logs()  # pyright: ignore[reportAttributeAccessIssue]
            except RuntimeError as e:
                if "Container is not initialized" in str(e):
                    logging.warning(f"Container already stopped, skipping log save: {e}")
                else:
                    raise
            container.remove()  # pyright: ignore[reportAttributeAccessIssue]
```